### PR TITLE
fixes #277 -- wrap 0-360 longitude

### LINKED ISSUE #277: Problems loading nc file from copernicus.eu
Hello, 

It seems I can't load the NC file download from copernicus.eu correctly. I post a small version of the file [here](http://www.filedropper.com/adaptorcamsregionalfcretrieve-15879390618606012-22402-6-64f0e215-609d-4ca6-b65c-a9ef7dc53370).

The file loads correctly only if I do

```
library(stars)
nc <- read_ncdf(my_file)
```
> no 'var' specified, using pm10_conc, pm2p5_conc
> other available variables:
>  longitude, latitude, level, time
> No projection information found in nc file. 
>  Coordinate variable units found to be degrees, 
>  assuming WGS84 Lat/Lon.
> stars object with 4 dimensions and 2 attributes
> attribute(s):
>  pm10_conc [µg/m3]   pm2p5_conc [µg/m3]
>  Min.   :  0.08376   Min.   : 0.05215  
>  1st Qu.:  4.79556   1st Qu.: 2.08035  
>  Median :  9.77700   Median : 4.39209  
>  Mean   : 14.13988   Mean   : 5.51724  
>  3rd Qu.: 18.06134   3rd Qu.: 7.46488  
>  Max.   :196.88487   Max.   :54.89545  
> dimension(s):
>           from  to offset delta    refsys point                values    
> longitude    1 700     NA    NA EPSG:4326    NA [700] 0.05,...,359.95 [x]
> latitude     1 420     NA    NA EPSG:4326    NA [420] 30.05,...,71.95 [y]
> level        1   1     NA    NA        NA    NA                     0    
> time         1   2      0    24        NA    NA                  NULL

But if I try `system.file(my_file, package = "stars") %>% read_ncdf()`, I get

>  Error in nc_meta.character(.x) : NetCDF source cannot be empty string

or if I specify the x/y variables `read_ncdf(my_file, curvilinear = c("longitude","latitude"))`

> no 'var' specified, using pm10_conc, pm2p5_conc
> other available variables:
>  longitude, latitude, level, time
> No projection information found in nc file. 
>  Coordinate variable units found to be degrees, 
>  assuming WGS84 Lat/Lon.
> Error in .check_curvilinear(c_v, var, meta$variable, curvilinear) : 
>   Specified curvilinear coordinates are not 2-dimensional.

Then once loaded with ` read_ncdf(my_file)` if I try to `plot(nc)`

> Error in image.default(dims[[dimx]], dims[[dimy]], ar, asp = asp, xlab = xlab,  : 
>   increasing 'x' and 'y' values expected

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
Sorry for all the whitespace edits -- no idea why RStudio did that. Actual change is here: https://github.com/r-spatial/stars/compare/master...dblodgett-usgs:master#diff-2ea6e7b6427bc6e307a3a2891628688fR515

I changed some tests that were using the deprecated `+init=epsg...` pattern. 

There is a small subset .nc file included -- it has this ncdump: 
```
ncdump test.nc 
netcdf test {
dimensions:
	latitude = 11 ;
	level = 1 ;
	longitude = 11 ;
	time = UNLIMITED ; // (1 currently)
variables:
	float latitude(latitude) ;
		latitude:long_name = "latitude" ;
		latitude:units = "degrees_north" ;
	float level(level) ;
		level:long_name = "level" ;
		level:units = "m" ;
	float longitude(longitude) ;
		longitude:long_name = "longitude" ;
		longitude:units = "degrees_east" ;
	float pm10_conc(time, level, latitude, longitude) ;
		pm10_conc:_FillValue = -999.f ;
		pm10_conc:species = "PM10 Aerosol" ;
		pm10_conc:units = "µg/m3" ;
		pm10_conc:value = "hourly values" ;
		pm10_conc:standard_name = "mass_concentration_of_pm10_ambient_aerosol_in_air" ;
	float time(time) ;
		time:long_name = "ANALYSIS time from 20200101" ;
		time:units = "hours" ;

// global attributes:
		:title = "PM10/PM25 Air Pollutant ANALYSIS at the Surface" ;
		:institution = "Data produced by Meteo France" ;
		:source = "Data from ENSEMBLE model" ;
		:history = "Tue Apr 28 08:16:38 2020: ncks -v pm10_conc -d longitude,245,255 -d latitude,210,220 -d time,1,1 adaptor.cams_regional_fc.retrieve-1587939061.8606012-22402-6-64f0e215-609d-4ca6-b65c-a9ef7dc53370.nc test.nc\n",
			"Model ENSEMBLE ANALYSIS" ;
		:ANALYSIS = "Europe, 20200101-20200102+[12H_12H]" ;
		:summary = "ENSEMBLE model hourly ANALYSIS of PM10/PM25 concentration at the Surface from 20200101-20200102+[12H_12H] on Europe" ;
		:project = "MACC-RAQ (http://macc-raq.gmes-atmosphere.eu)" ;
		:NCO = "netCDF Operators version 4.9.2 (Homepage = http://nco.sf.net, Code = http://github.com/nco/nco)" ;
data:

 latitude = 50.95, 50.85, 50.75, 50.65, 50.55, 50.45, 50.35, 50.25, 50.15, 
    50.05, 49.95 ;

 level = 0 ;

 longitude = 359.55, 359.65, 359.75, 359.85, 359.95, 0.05, 0.15, 0.25, 0.35, 
    0.45, 0.55 ;

 pm10_conc =
  13.04929, 13.03968, 12.66358, 11.87934, 11.30076, 10.99139, 10.85459, 
    11.36539, 11.95021, 11.59317, 11.4792,
  12.70476, 12.10586, 11.13051, 10.7099, 10.88381, 11.16908, 10.67309, 
    10.7201, 11.10359, 11.63326, 12.30173,
  12.42196, 11.08589, 10.83782, 10.39032, 10.65027, 11.00204, 10.49122, 
    10.40885, 10.3397, 10.45046, 12.00868,
  13.78302, 13.17633, 13.44591, 10.77862, 10.44847, 10.55761, 10.67039, 
    10.72496, 10.72496, 10.66675, 12.87772,
  15.72539, 15.78147, 14.95988, 13.75113, 13.92957, 13.86314, 13.36315, 
    12.98181, 12.64667, 12.63675, 12.2462,
  16.35883, 16.20965, 15.94017, 15.63903, 13.85155, 13.52367, 12.89621, 
    12.29233, 12.14261, 11.98558, 11.32733,
  16.18932, 16.0647, 15.69649, 15.32932, 15.01967, 12.70999, 12.2364, 
    12.12399, 12.08329, 11.52469, 11.69974,
  15.75275, 15.76169, 15.38595, 15.02883, 14.84504, 12.92787, 11.42234, 
    11.38142, 11.51878, 11.83727, 11.81897,
  14.97006, 15.08262, 14.90882, 14.71722, 14.72635, 12.23137, 11.37701, 
    11.28959, 11.54731, 11.7407, 11.16194,
  15.34175, 14.95471, 14.85687, 14.77505, 14.85427, 12.25936, 11.10328, 
    10.93136, 11.44804, 11.27015, 10.2371,
  15.51283, 15.37109, 14.86563, 14.88579, 12.9923, 12.2298, 11.48328, 
    10.66311, 10.61721, 10.44376, 9.632799 ;

 time = 36 ;
}
```

### CODE REVIEW

**Comment by dblodgett-usgs** [comment]
This is the trick here -- and characteristic of why @mdsumner is nervous about this kind of behavior. I didn't have the units available in this function so didn't add the degrees check. I can do a little more to make that check as well if you'd be more comfortable.

File: R/ncdf.R [hunk 1]
```diff
@@ -506,30 +512,43 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
...
+                               atts) {
+
   to_rectilinear = FALSE
+
+  ## hack longitudes for #277
+  if (any(lon_coord <- grepl("lon", coord_var$X, ignore.case = TRUE))) {
+    lon_coord <- coord_var$X[lon_coord][1]
+    lons <- coords[[lon_coord]]
+    if (!is.null(lons) && !regular_intervals(lons, epsilon = eps) &&
+        .is_degrees(atts, lon_coord) &&
```
Comment (dblodgett-usgs) [inline R/ncdf.R:526]: I factored out a .is_degrees function and call it here now.

File: R/ncdf.R [hunk 2]
```diff
@@ -236,17 +237,22 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
   }
 }
 
+.is_degrees <- function(atts, var) {
```
Comment (dblodgett-usgs) [inline R/ncdf.R:240]: This utility gets used in two places now.

File: R/ncdf.R (hunk 1 above)
Comment (edzer) [reply R/ncdf.R:526]: LGTM!

**Comment by dblodgett-usgs** [comment]
Forgot this PR was still open. I added a commit that also fixes #301. Down a bit of an ifelse rabbit hole but tests all pass.

**Comment by dblodgett-usgs** [comment]
@edzer -- anything I should be doing to get this PR across the finish line?

**Comment by edzer** [comment]
Thanks a lot, David!

**Pull request merged**